Register, Docker Compose, Fast API

hosseinjafar opened the pull request:
Thanks for the nice package! I use a docker container with Ubuntu 24 on it. running `apt install plastimatch` was good enough to get everything to work. no need to install the compiled binaries in python manually.

The following functionality was added along with documentation and unit tests:

## Register:
The register command to the Plastimatch is now implemented in pyplastimatch.py the tests are placed in test/test_register.py.

## Docker compose:
Simplifies volume mounting and environment setting when running a container.

## FastAPI:
I use pyplastimatch on a remote cluster as a server. Implemented pyplasti.py to call register using http post request.

hosseinjafar committed:
added submodule

hosseinjafar committed:
trying setup a clean docker file and installation script

hosseinjafar committed:
renamed docker folder

hosseinjafar committed:
got the image bulding to work. plastimatch command runs on the cli. it is to be tested from inside the python

hosseinjafar committed:
made some slolid progress on the registration. we ge the input now, let's finish calling plastimatch

hosseinjafar committed:
put down the draft way to call register on plasti match. let's test it now

hosseinjafar committed:
downloaded the sample data for testing of pyplastimatch

hosseinjafar committed:
wrote the first test

hosseinjafar committed:
runnign registration worked for the basic example case

hosseinjafar committed:
just organized the docker file a bit better

hosseinjafar committed:
wrote a simple api for registration

hosseinjafar committed:
wrote a bash script to run the api

hosseinjafar committed:
updated the input model in the api to handle paths to the temp_data

hosseinjafar committed:
added some print statement for the logs!

hosseinjafar committed:
put a .env file that helps out docker compose with environment variables

hosseinjafar committed:
Update README.md

renamed dockerfiles with docker_src since this folder contains more than dockerfile.

LennyN95 commented:
Thank you for your contribution @hosseinjafar.

The fast api implementation and updated docker file are interesting (and if they are of general interest to the community I suggest to make them publicly available in a separate repository that we can reference).

As in the current state, the Dockerfile and Compose setup are quite specific (e.g., only the registration command you add to the package is published in the fast api, no description how the referenced test files can be acquired, ..). 
 
The path to the Dockerfile also should not be changed unless for a specific reason to maintain compatibility where possible.

hosseinjafar commented:
Fair enough!
I got the test data for registration from plastimatch registration tutorial here:
https://plastimatch.org/registration_tutorial.html 

Best

hosseinjafar committed:
exporing transforms as nrrd

hosseinjafar committed:
made a bunch of changes to work with convert over the api. the changes include:\n 1. replacing _ with - in the inputs for the convert command.\n 2. automatically removing \temp\
temp_data/ from all the inputs and adding the correct path for temp_data.
implementing the api call for conver

hosseinjafar committed:
simplified the api and tested it over the web. good to go to the next stage

hosseinjafar committed:
very important bug fix. fastapi needs to be run with the run command and not the dev command for the other containers to notice it

hosseinjafar committed:
played with the temp file paths

hosseinjafar committed:
simpler docker i guess

hosseinjafar committed:
Merge branch 'main' of https://github.com/engerlab/pyplastimatch-brachyutils
